services/libs/tinybird/pipes/leaderboards_project_active_contributors.pipe

```diff
@@ -0,0 +1,68 @@
+DESCRIPTION >
+    Leaderboard ranking projects by number of active contributors. Counts unique members with
+    contribution activities (collaboration or code contributions). Compares the current period
+    (last 12 months) with the previous period (12-24 months ago).
+
+NODE leaderboards_project_active_contributors_projects
+DESCRIPTION >
+    Retrieves all projects from the populated datasource
+
+SQL >
+    SELECT id, name, slug, segmentId, logoUrl
+    FROM insights_projects_populated_ds
+    GROUP BY id, name, slug, segmentId, logoUrl
+
+NODE leaderboards_project_active_contributors_activity_types
+DESCRIPTION >
+    Filters activity types to include only collaboration and code contribution activities
+
+SQL >
+    SELECT activityType, platform FROM activityTypes FINAL WHERE isCollaboration or isCodeContribution
+
+NODE leaderboards_project_active_contributors_current_period
+DESCRIPTION >
+    Counts unique contributing members per project for the last 12 months
+
+SQL >
+    SELECT segmentId, uniq(memberId) as contributor_count
+    FROM activityRelations_deduplicated_cleaned_ds ar
+    INNER JOIN
+        leaderboards_project_active_contributors_activity_types at
+        ON ar.type = at.activityType
+        AND ar.platform = at.platform
+    WHERE timestamp >= now() - INTERVAL 12 MONTH AND timestamp < now() AND isContribution = 1
+    GROUP BY segmentId
```

Comment:
If you are not selecting data from the join table (`leaderboards_project_active_contributors_activity_types`)  it usually means that u don't need an inner join, and can get away with a semi-join which is faster most of the time.

`SELECT ar.segmentId, uniq(ar.memberId) AS contributor_count
FROM activityRelations_deduplicated_cleaned_ds AS ar
WHERE
  ar.timestamp >= now() - INTERVAL 12 MONTH
  AND ar.timestamp < now()
  AND ar.isContribution = 1
  AND (ar.type, ar.platform) IN (
    SELECT DISTINCT activityType, platform
    FROM leaderboards_project_active_contributors_activity_types
  )
GROUP BY ar.segmentId;`